resource.c

```diff
@@ -194,7 +198,9 @@ int res_read(int res_id, void *out, void *hint, int pid, int flags)
 			errno = err;
 			return -1;
 		}
-		sscanf(t.version, "%*s%*s%*s%[^\t\n]", (char *) out);
+		sscanf(t.version, "%*s%*s%*s%[^\t\n]", t.version);
```

Comment:
sscanf into t.version from t.version is strange.  Not sure offhand whteher sscanf guarantees that it won't overwrite it's working space.

Reply:
It worked, however this does not look clean, so I have changed this with a cleaner solution.